src/core/server/saved_objects/saved_objects_type_registry.ts

```diff
@@ -23,14 +23,17 @@ import { SavedObjectsType } from './types';
 /**
  * See {@link SavedObjectTypeRegistry} for documentation.
  *
- * @internal
- * */
-export type ISavedObjectTypeRegistry = PublicMethodsOf<SavedObjectTypeRegistry>;
+ * @public
+ */
+export type ISavedObjectTypeRegistry = Pick<
+  SavedObjectTypeRegistry,
+  'getType' | 'getAllTypes' | 'getIndex' | 'isNamespaceAgnostic' | 'isHidden'
+>;
 
 /**
  * Registry holding information about all the registered {@link SavedObjectsType | savedObject types}.
  *
- * @internal
+ * @public
  */
 export class SavedObjectTypeRegistry {
```

Comment:
I was 'forced' to expose this class as public to get proper doc generation for the associated `ISavedObjectTypeRegistry` interface. I'm starting to wonder if we really should use `Pick`-based interface when exposing public interfaces from internal implementations...